packages/chat-client/src/utils/jwtAuth.ts

```diff
@@ -0,0 +1,137 @@
+import { base58btc } from "multiformats/bases/base58";
+import bs58 from "bs58";
+import * as ed25519 from "@noble/ed25519";
+import { isValidObject } from "@walletconnect/utils";
+
+interface JwtHeader {
+  typ: string;
+  alg: string;
+}
+
+export interface InviteKeyClaims {
+  iss: string;
+  sub: string;
+  aud?: string;
+  ksu?: string;
+  pkh?: string;
+  pke?: string;
+  iat: number;
+  exp: number;
+}
+
+export const DAY_IN_MS = 86400 * 1000;
+
+export const DID_METHOD_KEY = "key";
+export const DID_DELIMITER = ":";
+export const DID_PREFIX = "did";
+export const DID_METHOD_PKH = "pkh";
+
+export const JWT_DELIMITER = ".";
+
+export const MULTICODEC_ED25519_HEADER = "K36";
+export const MULTICODEC_X25519_HEADER = "Jxg";
+
+const concatUInt8Arrays = (array1: Uint8Array, array2: Uint8Array) => {
+  const mergedArray = new Uint8Array(array1.length + array2.length);
+  mergedArray.set(array1);
+  mergedArray.set(array2, array1.length);
+
+  return mergedArray;
+};
+
+export const composeDidPkh = (accountId: string) => {
+  return `${DID_PREFIX}${DID_DELIMITER}${DID_METHOD_PKH}${DID_DELIMITER}${accountId}`;
+};
+
+export const jwtExp = (issuedAt: number) => {
+  return issuedAt + DAY_IN_MS;
+};
+
+const objectToHex = (obj: unknown) => {
+  if (!isValidObject(obj)) {
+    throw new Error(`Supplied object is not valid ${JSON.stringify(obj)}`);
+  }
+
+  // toString("base64url") isn't universally supported.
+  return Buffer.from(new TextEncoder().encode(JSON.stringify(obj)))
+    .toString("base64")
+    .replace(/=/g, "")
+    .replace(/\+/g, "-")
+    .replace(/\//g, "_");
```

Comment:
Could you explain the substitutions here pls/add contextual comments?

Also seems like this exact pattern is reused below in `encodeJwt`, might make sense to turn this into its own reusable util function.

Reply:
The reason is `toString("base64url")` isn't supported in every dev environment, eg it might work when run in node, but when built in vite and other inconsistencies. This just achieves what `base64url` already does, which is base64 encode the buffer, but instead of `+`, use `-` and instead of `/`  use `_` and remove any padding (`=`).

I'll write this into a comment